test/parallel/test-diagnostics-channel-module-import-error.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
```

Comment:
That's not how diagnostics_channel tests have been written though. That wouldn't match with the rest of the diagnostics_channel tests.

Reply:
I don’t think that’s a strong argument for continuing with the old pattern. Eventually it’ll be more important that things work correctly in ESM than in CommonJS, so we avoid creating tech debt for ourselves by writing tests in ESM now.

And if there are existing Diagnostics Channel tests in CommonJS, all the more reason to have new ones in ESM since we already have coverage for the “in CommonJS” case.

Reply:
> Eventually it’ll be more important

that's a pretty bold claim. if it's important to have full coverage, then tests should be authored in both (or evaluated in both), rather than excluding one for the other's sake.

Reply:
I also find the new test runner-style tests much easier to read, and we can take advantage of improvements it offers like the snapshot tool. Especially with regard to some of these tests that are asserting stack traces, those are ideal things to use the snapshot tool for so that future PRs don’t need to update unrelated tests every time stack traces change.

I think the best way forward would be to do a separate PR to refactor the existing tests that get modified in this one, to migrate them to the new style and snapshot tool, so that then this PR doesn’t need to edit “unrelated” tests to add Diagnostics Channel to the CommonJS loader. There are several contributors helping migrate the `out/message` tests into the new style who might be interested in helping if you want to open an issue listing the tests that need migrating.